src/libraries/System.Private.CoreLib/src/System/Threading/RegisteredWaitHandle.Windows.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+#if !FEATURE_WASM_THREADS
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
+#endif
+    public sealed partial class RegisteredWaitHandle : MarshalByRefObject
```

Comment:
Similarly to registered waits, timers in NativeAOT also use the Windows thread pool. It would make sense to move that portion to the libraries as well and fork it between the portable and Windows thread pool implementations, so that the appropriate timer implementation would be used based on `UseWindowsThreadPool` in all runtimes.